### ISSUE TITLE
WCAG 1.4.10 AA: Carousel functionality lost when site reflowed (Issue 18)	

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#373]
### Summary

Via Vision Australia assessment: August 2024

Impact: medium

Note: DTA have a 90 day remediation period to address the identified issues within the audit, all issues must be resolved to obtain WCAG 2.2 certification for digital.gov.au.

### Steps to reproduce

[digital.gov.au beta | digital.gov.au](https://www.digital.gov.au/) (https://www.digital.gov.au)

### Observed outcome

When the site is viewed at an equivalent of 320 x 256 CSS pixels (1280 x 1024 CSS pixels at 400%), content and buttons within the carousel becomes truncated. While those navigating via keyboard can bring the buttons and content into focus, they cannot be accessed by those navigating using a pointer device (such as a mouse, head wand, touch, or eye-tracking software).
Additionally, when viewing the site on a mobile device in landscape orientation, content in the slider is truncated.

**Slider links cannot be reached via pointer devices as the section is truncated and does not scroll vertically**
![Picture30](https://github.com/user-attachments/assets/1c9a0fad-c2ec-483a-9c8d-c2f3c777c60f)

**Full screen view of slider content when reflowed**
![Picture31](https://github.com/user-attachments/assets/9860107b-9841-4220-b12a-f086eccdd20a)
 
**Slider content is truncated in mobile landscape orientation**
![Picture32](https://github.com/user-attachments/assets/79d97042-7aea-432f-93a3-9c692e17e8b0)
 
### Code used
```
<div class="ct-slider__container" data-slider="">
<h3 class="ct-heading ct-theme-light ct-slider__title">
Featured initiatives
</h3>
<div class="ct-slider__slides" data-slider-panel="" style="height: 380px;">
<div class="ct-slider__slides__inner" data-slider-rail="" style="width: 1440px; height: 380px; left: 0%;">
<div class="ct-slide ct-theme-light ct-slide--image-left " data-slider-slide="" style="width: 720px; left: 0px; height: 380px;" data-slider-slide-hidden="true">
<div class="ct-slide__image">
<img class="ct-image ct-theme-light " src="/sites/default/files/images/2024-08/ai-policy-cover_square.jpg" alt="An image of the front cover of the Australian Government's policy for the responsible use of AI in government.">
</div>
<div class="ct-slide__wrapper">
<div class="ct-slide__inner">
<h2 class="ct-heading ct-theme-light ct-slide__title">
AI in government policy
</h2>
... 
</div>
</div>
... 
</div>
</div>
</div>
</div>
```
### Why this matters
Users with low vision who need to enlarge content may not have full access to the content and functionality of the slides. 

### Expected outcome

Preferably, consider the need for a carousel component. Carousels can present multiple barriers for people with disabilities and increase cognitive load. As the amount of content in this carousel is very small, it could be presented in another way, such as Call to Action (CTA) cards or as a list. 

Alternatively, ensure content can be zoomed to 400% on a 1280px wide monitor without any loss of content or functionality and without having to rely on horizontal scrolling. This can be achieved by implementing a responsive design that reflows the content according to screen size. It may also be necessary to offer the information in another way when the site is reflowed. 

### Related issues

For other issues related to this component, please refer to:

•	Issue 6 – Carousel not contained within landmark region
•	Issue 16 – Loss of carousel content when text resized
•	Issue 25 – Visually hidden carousel elements in mobile focus order
•	Issue 26 – Focus lost when carousel controls are disabled

**Comment by joshua-salsadigital** [comment]
Tested this with latest code from uikit. 
https://github.com/civictheme/uikit/commit/f972c4ca6209730ab0922a7d9895bce8cb83d864 
Front end reload fixes size on the slider. 

Tested on civic theme. Latest main issue non replicable. 

Tried to see if issue exits on live site https://www.digital.gov.au/  slider no loner enabled. 
Also the scroll lock for slider seem to be inference from Was this page helpful? Additional overlay 

@fionamorrison23

**Comment by daniel-grynenko** [comment]
Test env: DEV
Overall test status: PASS

Test results:
	1.	Tested slider at 400% zoom to check if content and buttons remain visible and accessible.
	2.	Verified functionality on mobile landscape to ensure no truncation or loss of content.
	3.	Checked navigation via keyboard and pointer devices for accessibility.
	4.	Confirmed reflow behaviour and assessed alternative layouts for usability.

Desktop view - 400% zoom:
<img width="500" alt="Image" src="https://github.com/user-attachments/assets/e3d782a9-6d22-4f9b-b595-6c7eb4ac5e10" />

Mobile view - portrait:
<img width="450" alt="Image" src="https://github.com/user-attachments/assets/113b4787-56f1-4701-b5a4-34222b61aa04" />

Mobile view - landscape:
<img width="1000" alt="Image" src="https://github.com/user-attachments/assets/61229d3f-ed79-4fd8-a13f-3b25cce92a6a" />
